pulsar-function-go/conf/conf.go

```diff
@@ -80,14 +80,34 @@ var (
 	confContent  string
 )
 
+func fileExists(path string) bool {
+	if path == "" {
+		return false
+	}
+	_, err := os.Stat(path)
+	if err != nil && os.IsNotExist(err) {
```

Comment:
the logic here is not correct, because it returns `true` when err != nil && ! os.IsNotExist(err).

Reply:
I think the logic is right, the check logic  is `err != nil && os.IsNotExist(err)`  instead of `err != nil && ! os.IsNotExist(err)`.

Reply:
after some doc reading, seems `os.Stat` check is redundant because `ioutil.ReadFile` did the same thing. The previous error is because we handle `ioutil.ReadFile` incorrect, so I did some change to the code, removed `fileExists` func, `Unmarshal` the yaml file only if `ioutil.ReadFile` returns no error. Also, if we use `os.Stat`, there might be some extreme condition that the file will have been changed between `os.Stat` and `ioutil.ReadFile` (ref: https://groups.google.com/forum/#!msg/golang-nuts/Ayx-BMNdMFo/4rL8FFHr8v4J)

Reply:
Yes, in here, we can't simply use `IsNotExist` to make judgments. There are many types of **PathError**, and the absence of a file is just one of them.

Reply:
```
func fileExists(path string) (bool, error) {
	_, err := os.Stat(path)
	if err == nil {
		return true, nil
	}
	if os.IsNotExist(err) {
		return false, nil
	}
	return false, err
}
```